crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3655,7 +3655,11 @@ impl<'db> TypeInferenceBuilder<'db> {
             // TODO: parse the expression and check whether it is a string annotation, since they
             // can be annotation expressions distinct from type expressions.
             // https://typing.readthedocs.io/en/latest/spec/annotations.html#string-annotations
-            ast::Expr::StringLiteral(_literal) => Type::Todo,
+            ast::Expr::StringLiteral(_literal) => {
+                let expr_id = expression.scoped_ast_id(self.db, self.scope());
+                self.types.expressions.insert(expr_id, Type::Todo);
+                Type::Todo
+            }
```

Comment:
If we do not add the expression type into `self.types.expressions` here (like we do for the other branches), it will later lead to a panic here:

https://github.com/astral-sh/ruff/blob/df45a0e3f966dd24feb0fece101dfa469ff24d1b/crates/red_knot_python_semantic/src/types.rs#L199